application/cache/cache.go

```diff
@@ -0,0 +1,266 @@
+func getOrCreateCacheEntryPrivate(ctx context.Context, organization models.Organization) (bool, map[string]api.RequestAbridged, error) {
+	var requestsMap map[string]api.RequestAbridged
+	// if a cache value does not exist, we create it
+	if err := cacheRead(ctx, PrivateRequestKeyPrefix+organization.Name, requestsMap); err != nil {
+		tx := models.Tx(ctx)
+
+		// RequestFilterParams is currently empty because the UI is not using it
+		filter := models.RequestFilterParams{}
+
+		requests := models.Requests{}
+		if err := requests.FindByOrganization(tx, organization, filter); err != nil {
+			return false, nil, err
+		}
+
+		requestsList, err := models.ConvertRequestsAbridged(ctx, requests)
+		if err != nil {
+			return false, nil, err
+		}
+		requestsMap := make(map[string]api.RequestAbridged)
+		for _, requestEntry := range requestsList {
+			requestsMap[requestEntry.ID.String()] = requestEntry
+		}
+
+		return true, requestsMap, cacheWrite(ctx, PrivateRequestKeyPrefix+organization.Name, requestsMap)
+	}
+
+	return false, requestsMap, nil
+}
+
+// Gets cache entry for organization, or creates one if none exists.
+// Returns true if new cache entry was created
+func getOrCreateCacheEntryPublic(ctx context.Context) (bool, map[string]api.RequestAbridged, error) {
+	var requestsMap map[string]api.RequestAbridged
+	// if a cache value does not exist, we create it
+	if err := cacheRead(ctx, PublicRequestKey, requestsMap); err != nil {
+		tx := models.Tx(ctx)
+
+		// RequestFilterParams is currently empty because the UI is not using it
+		filter := models.RequestFilterParams{}
+
+		requests := models.Requests{}
+		if err := requests.FindPublic(tx, filter); err != nil {
+			return false, nil, err
+		}
+		requestsList, err := models.ConvertRequestsAbridged(ctx, requests)
+		if err != nil {
+			return false, nil, err
+		}
+		requestsMap := make(map[string]api.RequestAbridged)
+		for _, requestEntry := range requestsList {
+			requestsMap[requestEntry.ID.String()] = requestEntry
+		}
+
+		return true, requestsMap, cacheWrite(ctx, PublicRequestKey, requestsMap)
+	}
+	return false, requestsMap, nil
+}
+
+func updateRequestPrivateCache(ctx context.Context, request models.Request, requestsMap map[string]api.RequestAbridged) error {
+	organization := request.Organization
+	_, cachedPrivately := requestsMap[request.UUID.String()]
+	// was cached privately, but request has been finished. Just need to remove from private cache.
+	if cachedPrivately && request.IsFinished() {
+		delete(requestsMap, request.UUID.String())
+		return cacheWrite(ctx, PrivateRequestKeyPrefix+organization.Name, requestsMap)
+	}
+	// was cached privately, but request visibility has changed to public.
+	// Need to remove from private and add to public cache.
+	if cachedPrivately && request.IsPublic() {
+		delete(requestsMap, request.UUID.String())
+		if err := cacheWrite(ctx, PrivateRequestKeyPrefix+organization.Name, requestsMap); err != nil {
+			return err
+		}
+	}
```

Comment:
It looks like adding `return nil` would work here and it might make it easier to understand what's going on.
(Same in the next function)

Reply:
If we changed the return err to return nil, how would we know when a call to cacheWrite failed?

Reply:
Sorry. I was referring  to the end of the outer `if` block, not to the `return err` that's already there.

In other words, if this outer `if` block is entered, does the code progression still need to continue into the rest of the function following the outer `if` block?  It's not that big of a deal, it's just that mentally for me it's easier to see that OK now we can forget the case where cachedPrivately && request.IsPublic.

Don't bother going back to change it. I just wanted to explain what I meant.  It's more of a personal preference than an actual issue.

Reply:
Thanks for the clarification. Now that the updateRequest<Type>Cache logic is in two functions instead of one, the code logic could actually be changed to an if .... else if .... else format, which would improve readability, and is probably worthwhile.